pyo3-build-config/src/impl_.rs

```diff
@@ -1280,13 +1306,36 @@ fn fixup_config_for_abi3(
 /// This must be called from PyO3's build script, because it relies on environment variables such as
 /// CARGO_CFG_TARGET_OS which aren't available at any other time.
 pub fn make_cross_compile_config() -> Result<Option<InterpreterConfig>> {
-    let mut interpreter_config = if let Some(paths) = cross_compiling_from_cargo_env()? {
-        load_cross_compile_config(paths)?
+    let env_vars = cross_compile_env_vars();
+
+    let host = cargo_env_var("HOST").ok_or("expected HOST env var")?;
+    let target = cargo_env_var("TARGET").ok_or("expected TARGET env var")?;
+
+    let target_arch =
+        cargo_env_var("CARGO_CFG_TARGET_ARCH").ok_or("expected CARGO_CFG_TARGET_ARCH env var")?;
+    let target_vendor = cargo_env_var("CARGO_CFG_TARGET_VENDOR")
+        .ok_or("expected CARGO_CFG_TARGET_VENDOR env var")?;
+    let target_os =
+        cargo_env_var("CARGO_CFG_TARGET_OS").ok_or("expected CARGO_CFG_TARGET_OS env var")?;
+
+    let interpreter_config = if env_vars.any() {
+        let cross_config =
+            cross_compiling_from_vars(&target_arch, &target_os, &target_vendor, env_vars)?;
```

Comment:
`target_os` and `target_vendor` are switched here. Maybe this would benefit from a struct with named fields similar to `CrossCompileEnvVars`?

Reply:
Added. Though I didn't change `cross_compiling` to take it as input because that's a public API. Think it's fine for now.